### PR TITLE
current code clean up

### PR DESCRIPTION
**Pull request by mateo-marinovic**

### CODE REVIEW

File: src/app/item-info/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import ItemCounter from "@/components/item-counter/item-counter";
+
+function ItemInfo() {
+  return (
+    <>
```
Comment (ilovric00) [inline src/app/item-info/[id]/page.tsx]: unnecessary fragment tags

File: src/app/item-info/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+      <ItemCounter></ItemCounter>
```
Comment (ilovric00) [inline src/app/item-info/[id]/page.tsx]: This can be self closing component

File: src/app/products/page.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"use client";
+import ItemList from "@/components/item-list/item-list";
+import { UsersProvider } from "@/contexts/users.context";
+import { ItemsHeader } from "@/components/items-header/items-header";
+
+export default function Products() {
+  return (
+    <>
```
Comment (ilovric00) [inline src/app/products/page.tsx:38]: unnecessary fragment tags

File: src/app/wish-list/page.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+function WishList() {
+  return (
+    <>
```
Comment (ilovric00) [inline src/app/wish-list/page.tsx:28]: unnecessary fragment tags

File: src/components/item-card/item-card.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    } else {
+      item.favoriteBy = item.favoriteBy.filter((x: number) => x !== id);
+    }
+
+    productsHttpClient.favoriteItem(item.id, item.favoriteBy);
+
+    setFavorite(!favorite);
+  };
+  return (
+    <>
```
Comment (ilovric00) [inline src/components/item-card/item-card.tsx]: unnecessary fragment tags

File: src/components/item-list/item-list.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  const list = items.map((item: Product, index: number) => {
+    return (
+      <ItemCard
+        key={index}
+        item={item}
+      ></ItemCard>
+    );
+  });
+  return (
+    <>
```
Comment (ilovric00) [inline src/components/item-list/item-list.tsx]: unnecessary fragment tags

File: src/components/item-counter/item-counter.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  }, [count]);
+
+  const incrementCount = useCallback(() => {
+    setCount(count + 1);
+  }, [count]);
+
+  const disable = count <= 0;
+
+  return (
+    <>
```
Comment (ilovric00) [inline src/components/item-counter/item-counter.tsx]: unnecessary fragment tags

File: src/components/item-list/item-list.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  const list = items.map((item: Product, index: number) => {
+    return (
+      <ItemCard
+        key={index}
+        item={item}
+      ></ItemCard>
+    );
+  });
+  return (
```
Comment (ilovric00) [inline src/components/item-list/item-list.tsx:7]: blank space before

File: src/components/items-header/items-header.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+"use client";
+import Link from "next/link";
+
+export function ItemsHeader() {
+  return (
+    <>
```
Comment (ilovric00) [inline src/components/items-header/items-header.tsx]: unnecessary fragment tags

File: src/components/items-header/items-header.tsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+"use client";
```
Comment (ilovric00) [inline src/components/items-header/items-header.tsx]: this doesn't need to be a client component

File: src/contexts/users.context.tsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+const userContextStateDefaultValue: UserContextState = {
+  id: 0,
+};
+
+const UsersContext = createContext<UserContextState>(
+  userContextStateDefaultValue
+);
+
+export function UsersProvider({ children }: UsersProviderProps) {
```
Comment (ilovric00) [inline src/contexts/users.context.tsx]: `export const UsersProvider: React.FC<PropsWithChildren> = ({ children }) => {...}`

File: src/types/product.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+export interface Product {
+  id: number;
+  title: string;
+  price: number;
+  image: string;
+  favoriteBy: number[];
+}
```
Comment (ilovric00) [inline src/types/product.ts]: since this is an interface, a file should be located inside `src/interfaces/`

**Review by ilovric00 — CHANGES_REQUESTED** [review]

**Comment by mateo-marinovic** [comment]
I added search bar and fixe PR comments. Also, I accidentally pushed new feature in the same branch. In the future will create new branches for the new features

**Comment by ilovric00** [comment]
Too huge PR

File: src/components/items-header/items-header.tsx [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import Link from "next/link";
+
+export function ItemsHeader() {
+  return (
+    <Link href={"/wish-list"}>
```
Comment (ilovric00) [inline src/components/items-header/items-header.tsx:5]: `<Link href="/wish-list">`

File: src/components/item-list/item-list.tsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import ItemCard from "../item-card/item-card";
+import { Product } from "@/interfaces/product";
+interface ItemListProps {
+  items: Product[];
+}
+function ItemList({ items }: ItemListProps) {
+  return (
+    <ul className="">
```
Comment (ilovric00) [inline src/components/item-list/item-list.tsx:8]: empty className, remove it

**Review by ilovric00 — APPROVED** [review]

**Pull request merged**